chia/data_layer/data_layer_server.py

```diff
@@ -0,0 +1,60 @@
+import logging
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any, Dict
+
+from aiohttp import web
+
+from chia.data_layer.download_data import is_filename_valid
+from chia.server.upnp import UPnP
+from chia.util.path import path_from_root
+
+
+@dataclass
+class DataLayerServer:
+    root_path: Path
+    config: Dict[str, Any]
+    log: logging.Logger
+
+    async def start(self) -> None:
+        self.log.info("Starting Data Layer Server.")
+        self.port = self.config["host_port"]
+
+        # Setup UPnP for the data_layer_service port
+        self.upnp: UPnP = UPnP()
```

Comment:
Does the UPnP library we are using allow separate instances of the object without causing conflicts?

Reply:
I don't know.  While the data layer file propagation service 'structure' is in flux, it does not and will not exist in the full node process.  Is that what you were concerned about?  Or concerned about two instances even in separate processes on the same computer?

Reply:
I was concerned even in the separate process case, in the unlikely case that they both choose the same port. But otherwise, I think separate process handles all the other potential issues.

Reply:
I may not know enough about how UPnP works.  The processes pick their own ports as normal, which will be different for different services by design.  Separately UPnP connects to the router to try to request the port forwarding from service's port on the router to the service's port on the local system.  Is one of those ports what you were worried about colliding?  Or is there another port involved in UPnP.